examples/docker-rootful.yaml

```
@@ -44,16 +44,64 @@ provision:
     #!/bin/bash
     set -eux -o pipefail
     command -v docker >/dev/null 2>&1 && exit 0
-    if [ ! -e /etc/systemd/system/docker.socket.d/override.conf ]; then
-      mkdir -p /etc/systemd/system/docker.socket.d
+    readonly override_conf=/etc/systemd/system/docker.socket.d/override.conf
+    if [ ! -e "$override_conf" ]; then
+      mkdir -p $(dirname "$override_conf")
       # Alternatively we could just add the user to the "docker" group, but that requires restarting the user session
-      cat <<-EOF >/etc/systemd/system/docker.socket.d/override.conf
-      [Socket]
-      SocketUser={{.User}}
+      cat <<EOF >"$override_conf"
+    [Socket]
+    SocketUser={{.User}}
     EOF
     fi
     export DEBIAN_FRONTEND=noninteractive
     curl -fsSL https://get.docker.com | sh
+- mode: user  # configure docker under non-root user
+  script: |
+    #!/bin/bash
+    set -eux -o pipefail
+    command -v jq &>/dev/null || sudo apt-get install -y jq
+    readonly rootless_installed=$(systemctl --user list-unit-files docker.service &>/dev/null && echo true || echo false)
```

I think our templates exist not only to implement the required functionality, but also serves as documentation and tutorials for others to adapt for their own needs.

Therefore I think we should not aim for the most compact representation, but instead write them in a simplistic style aimed at beginners, who may not be very experienced in shell scripting.

This means using long-form options whenever possible[^1], and using simplified `if … then … else … fi` control flow instead of trying to squeeze as much functionality as possible into the least amount of lines.

[^1]: I know we currently don't do that, but I would be in favour of expanding options on already existing code as well. E.g. `curl -fsSL` is non-obvious to many people.

So I would prefer something like this:

```suggestion
    set -o errexit -o nounset -o xtrace -o pipefail

    if ! command -v jq &>/dev/null; then
        sudo apt-get install --assume-yes jq
    fi
    if systemctl --user list-unit-files docker.service &>/dev/null; then
        readonly rootless_installed=true
    else
        readonly rootless_installed=true
    fi
```